crates/host/src/lib.rs

```diff
@@ -130,20 +128,21 @@ pub async fn fetch_actor(
 }
 
 /// Fetch a provider from a reference.
-#[instrument(skip(provider_ref, link_name))]
+#[instrument(skip(registry_config, provider_id, host_id), fields(provider_ref = %provider_ref.as_ref()))]
```

Comment:
Nit: Might be worth having the provider ID as a field for easier log searching later